src/core/federation.ts

```diff
@@ -836,13 +836,12 @@ export class FederatedApp extends App {
         return true
     }
 
-    protected _advanceTime(nextTag: Tag) {
+    protected _iterationComplete(nextEvent?: TaggedEvent<Present>): void {
         let currentTime = this.util.getCurrentLogicalTime()
-        if (currentTime.isEarlierThan(nextTag.time)) {
+        if (nextEvent === undefined || currentTime.isEarlierThan(nextEvent.tag.time)) {
```

Comment:
@Soroosh129: is a LTC expected at _any_ conclusions of a logical time? The inherited logic here only sends a `TAG` _only if_ the next tag has a greater time value (i.e., it doesn't send a TAG when only the microstep is different). Is this correct? I wasn't sure.

Reply:
Microsteps are important under centralized coordination.
There should be an LTC message sent at the conclusion of each tag.

Reply:
OK, then this if-statement is wrong and should be removed. It causes `LTC` message to only be sent when there is a timevalue increment (as opposed to a microstep increment).